src/gfx/context_data.cpp

```diff
@@ -15,10 +20,14 @@ void ContextData::unbindFromContext() {
 }
 
 void ContextData::releaseContextDataConditional() {
+  ContextData::globalMutex.lock();
+
   if (context) {
     releaseContextData();
     unbindFromContext();
   }
+
+  ContextData::globalMutex.unlock();

```

Comment:
What's this global thing here?

Reply:
Don't like global things 😄

Reply:
Yeah temporary, I think I want to replace this whole ContextData concept by using the entt registry as storage.
It's basically a way to detach the storage of device data from the description (meshes, textures, etc.)